src/flex.skl

```diff
@@ -1609,7 +1609,8 @@ int yyFlexLexer::yy_get_next_buffer()
 	/* First move last chars to start of buffer. */
 	number_to_move = (int) (YY_G(yy_c_buf_p) - YY_G(yytext_ptr) - 1);
 
-	memcpy(dest,source,number_to_move);
+	if (number_to_move > 0)
```

Comment:
Is there any condition into which number_to_move can be negative? By looking at YY_DECL() code, it seems very very unlikely to me...

you could probably get away with changing the number_to_move type to size_t but the only reason why it isn't done, it is because all the other variables that number_to_move is used with are of type int.

There is 2 things that I would do:
1. Be absolutly sure that number_to_move is never negative (maybe with the help of an ASSERT())
2. Remove the positive test and simply use a typecase to size_t in the memcpy() function for number_to_move param. (but I really really think that it should be a memmove call(). If you check glibc source code, you'll find out that memmove simply take care of the overlapping region + internally use memcpy().